drv/sidecar-seq-server/src/front_io.rs

```diff
@@ -8,39 +8,41 @@ use drv_sidecar_front_io::{controller::FrontIOController, phy_smi::PhySmi};
 
 #[allow(dead_code)]
 pub(crate) struct FrontIOBoard {
-    pub fruid: I2cDevice,
     pub controllers: [FrontIOController; 2],
-    pub state_reset: bool,
     fpga_task: userlib::TaskId,
     auxflash_task: userlib::TaskId,
-    i2c_task: userlib::TaskId,
 }
 
 impl FrontIOBoard {
     pub fn new(
         fpga_task: userlib::TaskId,
-        i2c_task: userlib::TaskId,
         auxflash_task: userlib::TaskId,
     ) -> Self {
         Self {
-            fruid: i2c_config::devices::at24csw080_front_io(i2c_task)[0],
             controllers: [
                 FrontIOController::new(fpga_task, 0),
                 FrontIOController::new(fpga_task, 1),
             ],
-            state_reset: false,
             fpga_task,
             auxflash_task,
-            i2c_task,
         }
     }
 
-    pub fn phy_smi(&self) -> PhySmi {
+    pub fn phy(&self) -> PhySmi {
         PhySmi::new(self.fpga_task)
     }
 
-    pub fn present(&self) -> bool {
-        At24Csw080::validate(&self.fruid).unwrap_or(false)
+    pub fn present(i2c_task: userlib::TaskId) -> bool {
+        let fruid = i2c_config::devices::at24csw080_front_io(i2c_task)[0];
+        At24Csw080::validate(&fruid).unwrap_or(false)
+    }
+
+    pub fn initialized(&self) -> Result<bool, FpgaError> {
+        Ok(self
+            .controllers
+            .iter()
+            .map(|c| c.ready().unwrap_or(false))
+            .fold(true, |init, ready| init && ready))
```

Comment:
```suggestion
        Ok(self.controllers.iter().all(|c| c.ready().unwrap_or(false)))
```
(also, is it okay to discard errors from `ready()` here?)

Reply:
Ah nice, thank you.

The function is used in two places; 1) determine of the bitstream may need to be loaded in the controllers, which performs a more detailed check, and 2) to gate the `transceiver` task and the sequencing decision to start Tofino until the sequencer is sure a future power cycle of the front IO board is not needed. In both cases we're really only interested in the happy answer and take the conservative path (default to false) otherwise.

I realized the function signature does not reflect this so it now returns a `bool` instead of a `Result`.